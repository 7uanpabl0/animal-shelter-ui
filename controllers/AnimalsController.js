import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            url: `animal`,
            baseURL: process.env.API,
            data: animal,
            method: 'POST'
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    }
}

