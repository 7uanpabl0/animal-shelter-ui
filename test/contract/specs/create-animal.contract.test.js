import { provider } from '../config/initPact';
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

describe('Given An Animal service', () => {
    describe('Cuando se pide la creaciÃ³n de un animal', () => {


        var animal = {
            'name':'Lucas',
            'breed':'Catico',
            'gender':'Male',
            'isVaccinated':true,
            'vaccines': []}


        
        


        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'create animal',
                uponReceiving: 'a request to create an animal',
                withRequest: {


                    method: 'POST',
                    path: '/animals',
                
                    body: animal
                    /* Matchers.eachLike({
                        name: Matchers.string('Lucas'),
                        breed: Matchers.like("Beagle"),
                        gender: Matchers.like("Male"),
                        vaccinated: Matchers.boolean(true), */
                },

                
                willRespondWith: {
                    status: 201,
                    body: Matchers.like(animal),
                    
                    headers: {
                        'Content-Type': 'application/json'
                    },
                }
            });
        });

        it("It should then return the data of the animal created correctly", async() =>{
            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
});