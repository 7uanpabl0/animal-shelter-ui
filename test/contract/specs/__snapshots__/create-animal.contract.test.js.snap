// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal service Cuando se pide la creaci√≥n de un animal It should then return the data of the animal created correctly 1`] = `
Object {
  "breed": "Catico",
  "gender": "Male",
  "isVaccinated": true,
  "name": "Lucas",
  "vaccines": Array [],
}
`;
